libs/@local/graph/sdk/typescript/src/entity-type.ts

```diff
@@ -0,0 +1,25 @@
+import type { VersionedUrl } from "@blockprotocol/type-system";
+import type { GraphApi } from "@local/hash-graph-client";
+
+import type { AuthenticationContext } from "./authentication-context.js";
+
+/**
+ * Returns whether the user can instantiate the given entity types.
+ *
+ * The returned array has the same length and order as the input array of `entityTypeIds`.
+ * Each boolean value at index `i` indicates whether the user is permitted to instantiate the
+ * entity type at `entityTypeIds[i]`.
+ */
+export const canInstantiateEntityTypes = <
+  T extends readonly [VersionedUrl, ...VersionedUrl[]],
+>(
+  graphAPI: GraphApi,
+  authentication: AuthenticationContext,
+  entityTypeIds: T,
+): Promise<{ [K in keyof T]: boolean }> =>
```

Comment:
This would be better as a map of the ids to a boolean, so that the return could be used in locations which are decoupled from the calling function, but this works for now (the caller can always construct such a map)